osgp/shared/osgp-throttling-client/src/test/java/org/opensmartgridplatform/throttling/ThrottlingClientTest.java

```diff
@@ -168,6 +185,47 @@ void clientRequestsPermitByNetworkSegment() {
         .isEqualTo(Optional.of(expectedPermit));
   }
 
+  @Test
+  void unregisteredClientRequestsPermitByNetworkSegment() {
+    final short throttlingConfigId = 37;
+    final int clientId = 347198;
+    final int baseTransceiverStationId = 983745;
+    final int cellId = 2;
+    final int requestId = 894;
+    this.whenTheThrottlingClientUsesNextRequestId(requestId);
+    this.whenTheThrottlingServiceGrantsTheRequestedPermit(
+        throttlingConfigId, clientId, baseTransceiverStationId, cellId, requestId);
+
+    final Permit expectedPermit =
+        new Permit(throttlingConfigId, clientId, requestId, baseTransceiverStationId, cellId, null);
+
+    final Optional<Permit> requestedPermit =
+        this.throttlingClient.requestPermit(baseTransceiverStationId, cellId);
+
+    assertThat(requestedPermit)
+        .usingRecursiveComparison()
+        .ignoringExpectedNullFields()
+        .isEqualTo(Optional.of(expectedPermit));
+  }
+
+  @Test
+  void registerFailureClientRequestsPermitByNetworkSegment() {
+    final short throttlingConfigId = 37;
+    final int clientId = 347198;
+    final int baseTransceiverStationId = 983745;
+    final int cellId = 2;
+    final int requestId = 894;
+    this.whenTheThrottlingServiceReturnsFailureOnRegistration();
+
+    final Permit expectedPermit =
+        new Permit(throttlingConfigId, clientId, requestId, baseTransceiverStationId, cellId, null);
```

Comment:
The expected permit is not used and had better be removed, as the test verifies that the requested permit is not present.
The `throttlingConfigId`, and `clientId` don't belong in this test as these are only returned if the throttling service is up and functioning, which it is not in this scenario.
If you want to keep the `requestId` you had better add a call to `whenTheThrottlingClientUsesNextRequestId`, but it may be better to leave both out as the actual `requestId` appears to have no particular relevance to this scenario.

Reply:
Applied changes